### PR TITLE
Fix: Cannot read properties of undefined (reading 'children')

### PR DESCRIPTION
**Pull request by bqrkhn**
Adding test case for when we return an empty object, then graffy should be able to encode it. Currently, this throws the error `Cannot read properties of undefined (reading 'children')`

### CODE REVIEW

File: src/common/coding/encodeTree.js [hunk 1]
```diff
@@ -16,14 +16,15 @@ function encode(value, { version, isGraph } = {}) {
...
+      : null;
+
     // prettier-ignore
-    let children =
-      !isEmpty(props) ? makeNode(
-        range ? [{ $key: range, ...props}] : props, undefined, $ver
-      ).children :
-      isDef($chi) ? makeNode(
-        range ? [{ $key: range, $chi }] : $chi, undefined, $ver
-      ).children :
```
Comment (aravindet) [inline src/common/coding/encodeTree.js:26]: Previously it was assumed that if props wasn't empty, makeNode will return something.

But if props is something like `{ foo: {} }`, makeNode (correctly) returns undefined.

When this happens on a node with $ref, this code path is taken.

File: src/common/coding/test/roundTrip.test.js [hunk 2]
```diff
@@ -264,6 +264,29 @@ describe('graph', () => {
...
+          },
+        ],
+      },
+      {
+        person: [
+          {
+            $key: { example: 'filter', $cursor: ['something'] },
+          },
+        ],
+      },
```
Comment (bqrkhn) [inline src/common/coding/test/roundTrip.test.js:285]: Is this 2nd element of this array required? It wasn't in the original test case. What is the use of this if I may ask?

Comment (aravindet) [reply src/common/coding/test/roundTrip.test.js:285]: It's not a second element in the array, it's a second argument to the roundTrip function.

Roundtrip (defined above) is called with (original, expected, extraAssertions). We're passing expected, because here the expected value after encode & decode is slightly different from original (basically, $ref becomes a non-enumerable property).
